### ISSUE TITLE
HIGH: Hardcoded vocab size (262272) in logprob computation breaks cascade routing

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#733]
## Summary

The cascade/adaptive strategy computes logprobs with a hardcoded vocab size (262272) and hidden dim (2048) in `tools/unified_server.cpp`. If the loaded model uses a different vocab size (32000 for Qwen, 151936 for Zaya), the logits buffer overflows or reads past array bounds.

## The Bug

```cpp
int vs = 262272, hs = 2048;  // HARDCODED
```

This is used to size the logits buffer for the `forward() + lm_head()` path that computes real per-token confidence. If the model's vocab is smaller, the buffer wastes memory. If larger, it overflows. Correct logprob is essential for cascade/adaptive routing decisions.

## Impact

- Cascade/adaptive strategy makes wrong routing decisions with wrong model dimensions
- Potential buffer overflow with larger models
- The per-token strategy routing is defeated

## Location

`tools/unified_server.cpp` — inside `generate_completion()` lambda

**Comment by bong-water-water-bong** [comment]
All fixes merged to main via PR #802.
